url0e7d57473a_kyorohiro_HetimaUtil_tgz-pJ8-net_hetimatan_net_http_HttpServer3xxJ8/src/net/hetimatan/io/filen/CashKyoroFileHelper.java

```diff
@@ -0,0 +1,106 @@
+package net.hetimatan.io.filen;
+
+import java.io.File;
+import java.io.IOException;
+import java.io.OutputStream;
+
+import net.hetimatan.io.file.KyoroFile;
+import net.hetimatan.util.event.GlobalAccessProperty;
+
+
+public class CashKyoroFileHelper {
+
+	public final static String GATAG_DEFAULTCASH = "my.tmp";
+	private static int sID = 0;
+
+	public static File newCashFile() {
+		boolean deleted = true;
+		File base = null;
+		do {
+			sID++;
+			GlobalAccessProperty prop = GlobalAccessProperty.getInstance();
+			String pathAsString = prop.get(GATAG_DEFAULTCASH, System.getProperty("java.io.tmpdir"));
+			File dir = new File(pathAsString);
+			base = new File(dir, "virtualFile_"+sID);
+			if(base.exists()) {
+				deleted = base.delete();
+			} else {
+				deleted = true;
+			}
+		}while(!deleted);
+		return base;
+	}
+
+	public static byte[] newBinary(KyoroFile src) throws IOException {
+		return newBinary(src, 0);
+	}
+
+	public static byte[] newBinary(KyoroFile src, long start) throws IOException {
+		int len = (int)(src.length()-start);
+		if(len<=0) {
+			return new byte[0];
+		}
+		byte[] buffer = new byte[len];
+		long fp = src.getFilePointer();
+		try {
+			src.seek(start);
+			src.read(buffer);
+		} finally {
+			src.seek(fp);
+		}
+		return buffer;
+	}
+
+	public static void xcopy(File[] srcs, CashKyoroFile dest) throws IOException {
+		for(int i=0;i<srcs.length;i++) {
+			CashKyoroFile src = new CashKyoroFile(srcs[i], 16*1024, 10);
```

Comment:
Recommendation generated by Amazon CodeGuru Reviewer. Leave feedback on this recommendation by replying to the comment or by reacting to the comment using emoji.

**Problem**
Analysis of this package determined that this line of code contains a resource that might not have closed properly. A resource leak can slow down or crash your system.

**Fix**
Consider closing the following resource in a try-finally block: *src*. The resource is referenced at lines 59, 62. The resource is closed at line 63. There are other execution paths that do not contain closure statements, for example, when CashKyoroFile.seek throws an exception.

**More info**
[View resource management guidelines at oracle.com](https://www.oracle.com/technetwork/articles/java/trywithresources-401775.html) (external link).